src/semver/Version.d

```diff
@@ -0,0 +1,139 @@
+/*******************************************************************************
+
+    Utility to simplify working with semantic versioning entities, including
+    parsing and comparison.
+
+*******************************************************************************/
+
+module semver.Version;
+
+/**
+    Describes SemVer aware version instance
+ **/
+struct Version
+{
+    /// major version number
+    int major;
+
+    /// minor version number
+    int minor;
+
+    /// patch version number
+    int patch;
+
+    /// original version string
+    string representation;
+
+    /**
+        Returns:
+            standard string representation of the version
+     **/
+    string toString ( ) const pure
+    {
+        return this.representation;
+    }
+
+    /**
+        Parses version string of form vX.Y.Z
+
+        If string doesn't match the pattern, it is simply stored as
+        as representation without initializing any numeric version fields.
+
+        Params:
+            ver = string to parse
+
+        Returns:
+            matching Version struct
+     **/
+    static Version fromString ( string ver )
+    {
+        import std.exception : enforce;
+        import std.conv;
+        import std.regex;
+
+        static verRegex = regex(r"^v(\d+)\.(\d+)\.(\d+)$", "g");
```

Comment:
Even when we use always `vX.Y.Z`maybe it would be good to be more flexible and accept strings without the leading `v` too (`regex(r"^v?(\d+)\.(\d+)\.(\d+)$", "g");`). Maybe it can accept a `bool` for strict mode.